src/readers/reader_tum.py

```diff
@@ -0,0 +1,101 @@
+import open3d as o3d
+import mrob
+import numpy as np
+import os
+import cv2
+
+from reader import Reader
+from nptyping import NDArray, Shape, Float, UInt8
+from pathlib import Path
+from src.database import Database
+
+
+class ReaderTUM(Reader):
+    def __init__(self, path_to_dataset: str, max_difference: int = 1000):
+        self.path_to_dataset = path_to_dataset
+        self.max_difference = max_difference
+
+    def read_dataset(self) -> Database:
+        return super().read_dataset()
+
+    def _get_images_pcds_traj(
+        self,
+    ) -> tuple[list[str], list[str], list[NDArray[Shape["4, 4"], Float]]]:
+        depth_with_timestamps = self.__read_folder(
+            os.path.join(self.path_to_dataset, "depth")
+        )
+        rgb_with_timestamps = self.__read_folder(
+            os.path.join(self.path_to_dataset, "rgb")
+        )
+        matches = self.__associate(rgb_with_timestamps, depth_with_timestamps)
+        timestamps, traj = self.__read_trajectory()
+        timestamps = np.asarray(timestamps)
+        images_paths = []
+        pcds_paths = []
+        res_traj = []
+        for (a, b) in matches:
+            diff = abs(timestamps - np.mean([a, b]))
+            min_diff_ind = np.argmin(diff)
+            images_paths.append(rgb_with_timestamps[a])
+            pcds_paths.append(depth_with_timestamps[b])
+            res_traj.append(traj[min_diff_ind])
+        return images_paths, pcds_paths, res_traj
+
+    @staticmethod
+    def _get_rgb(path_to_image: str) -> NDArray[Shape["*, *, 3"], UInt8]:
+        return cv2.imread(path_to_image, cv2.IMREAD_COLOR)
+
+    @staticmethod
+    def _get_pcd(path_to_pcd: str) -> o3d.geometry.PointCloud:
+        depth_image = cv2.imread(path_to_pcd, cv2.IMREAD_ANYDEPTH)
+        depth_image = o3d.geometry.Image(depth_image)
+        intrinsic_matrix = np.asarray([[525.0, 0, 319.5], [0, 525.0, 239.5], [0, 0, 1]])
+        intrinsic = o3d.camera.PinholeCameraIntrinsic()
+        intrinsic.intrinsic_matrix = intrinsic_matrix
+        intrinsic.width, intrinsic.height = 640, 480
+        pcd = o3d.geometry.PointCloud.create_from_depth_image(
+            depth_image, intrinsic, depth_scale=5000
```

Comment:
same

Reply:
also better to use custom method, not open3d one

Reply:
are you sure that open3d method is broken?